### PR TITLE
refactor: use CML utilities were applicable

### PR DESCRIPTION
**Pull request by littlespex**
Refactor the plugin to use more of the Common Media Library's CMCD utilities.

### CODE REVIEW

File: src/cmcd.js [hunk 1]
```diff
@@ -0,0 +1,324 @@
+import { CmcdObjectType } from '@svta/common-media-library/cmcd/CmcdObjectType';
+import { CmcdStreamType } from '@svta/common-media-library/cmcd/CmcdStreamType';
+import { CmcdStreamingFormat } from '@svta/common-media-library/cmcd/CmcdStreamingFormat';
+
+export class Cmcd {
```
Comment (littlespex) [inline src/cmcd.js]: The CML handles assignment of CMCD keys to the correct header, so there is no need to divide them into separate classes. This new `Cmcd` class is just a consolidation of the previous `cmcdKeys` classes.

File: src/plugin.js [hunk 2]
```diff
@@ -102,54 +85,6 @@ class Cmcd extends Plugin {
 
 }
 
-function generateUuid() {
```
Comment (littlespex) [inline src/plugin.js:105]: CML exports a `uuid` function specifically for use in generating session and content IDs.

File: src/plugin.js [hunk 3]
```diff
@@ -102,54 +85,6 @@ class Cmcd extends Plugin {
...
-      return `${key}`;
-    }
-    return `${key}=${obj[key]}`;
-
-  });
-
-  return keyValPair.filter(s => !s.includes('undefined')).join(',');
-}
-
-function toHeader(cmcdHeaders) {
```
Comment (littlespex) [inline src/plugin.js:140]: CML has a header generation utility that handles mapping the keys to the correct header shards.

File: package.json [hunk 4]
```diff
@@ -75,7 +75,7 @@
   },
   "dependencies": {
     "@montevideo-tech/cmcd-validator": "^0.3.1",
-    "@svta/common-media-library": "^0.4.2",
+    "@svta/common-media-library": "^0.4.5",
```
Comment (littlespex) [inline package.json]: Update the dependency for minor bug fixes and improvements.

File: src/cmcdKeys/common.js [hunk 5]
```diff
@@ -1,39 +0,0 @@
-export const roundedToNearstHundredth = (number) => {
```
Comment (littlespex) [inline src/cmcdKeys/common.js:1]: CML will correctly round the bitrate values, so these functions are no longer needed.

File: src/cmcd.js (hunk 1 above)
Comment (joaquinbartaburu-wowza) [inline src/cmcd.js:5]: The class should have a different name than the plugin class itself.

File: src/plugin.js [hunk 6]
```diff
@@ -1,11 +1,8 @@
 import videojs from 'video.js';
 import {version as VERSION} from '../package.json';
-import { CmcdRequest } from './cmcdKeys/cmcdRequest';
-import { CmcdObject } from './cmcdKeys/cmcdObject';
-import { CmcdSession } from './cmcdKeys/cmcdSession';
-import { CmcdStatus } from './cmcdKeys/cmcdStatus';
+import { appendCmcdHeaders } from '@svta/common-media-library/cmcd/appendCmcdHeaders';
 import { appendCmcdQuery } from '@svta/common-media-library/cmcd/appendCmcdQuery';
-import { keyTypes } from './cmcdKeys/common.js';
+import { uuid } from '@svta/common-media-library/utils/uuid';
```
Comment (joaquinbartaburu-wowza) [inline src/plugin.js:6]: The class should be imported.

**Review by joaquinbartaburu-wowza — CHANGES_REQUESTED** [review]
.

File: src/plugin.js (hunk 6 above)
Comment (littlespex) [reply src/plugin.js:6]: What class?

File: src/cmcd.js (hunk 1 above)
Comment (JoaquinBCh) [reply src/cmcd.js]: The class should have a different name than the plugin class itself.

File: src/plugin.js (hunk 6 above)
Comment (JoaquinBCh) [inline src/plugin.js:7]: The new cmcd class should be imported.

**Review by JoaquinBCh — CHANGES_REQUESTED** [review]
Hi Casey, I used the wrong account to make the PR review 😅.  
I added some comments in the code.
Also I made those changes and I tested it using the cmcd-validator and I'm getting some things wrong.
- I saw that on the first request MTP is sending NaN value and if it's NaN should not be sended.
image
-  Sometimes I saw that the request is sending multiple times cmcd query param.
image
- The last thing is that nor is sended as a non relative path.
image
Feel free to reach us If you have any question.
Cheers.

**Comment by littlespex** [comment]
@JoaquinBCh 

The `NaN` issue is a bug in CML which will be fixed on the next release (later today). 

The other issues seem like they would have existed before this PR, as there is no code in the previous version of CML to handle `nor`, and the `appendCmcdQuery` function hasn't changed in the latest version. Does the current version of the videojs plugin exhibit these issues?

**Comment by coatesjuan** [comment]
Hi @littlespex ! About the appendCmcdQuery() you are right, we just checked and it seems to be a problem that we already had as you guessed, thanks for the catch 😄.  And regarding the nor bug, we re-tested and after the last commit it is working fine.
We will be waiting for the svta release, thanks!!

**Comment by littlespex** [comment]
CML has been updated with a fix for the `NaN` issue, and it should also take care of the duplicate `CMCD` query params.

**Review by coatesjuan — APPROVED** [review]

**Review by JoaquinBCh — APPROVED** [review]

**Pull request merged**